### PR TITLE
refactor(m-upgrade): Refactor owner reference fetch logic in m-upgrade

### PR DESCRIPTION
**Pull request by imazik**
- Refactors owner reference fetch logic in m-upgrade
- Change environment variable name  
- Change `jsonkey` in remote exec output

#### This is updated config & corresponding upgrade job
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: pool-upgrade-config
  namespace: default
data:
  upgrade: |
    casTemplate: cstor-pool-081-082
    resources:
    - name: pool-c3jd
      kind: cStorpool
      namespace: openebs
    - name: pool-ji38
      kind: cStorpool
      namespace: openebs
    - name: pool-jkd4
      kind: cStorpool
      namespace: openebs
---
apiVersion: batch/v1
kind: Job
metadata:
  name: spc-cstor-sparse-pool-upgrade
spec:
  template:
    spec:
      serviceAccountName: super-admin
      containers:
      - name:  upgrade
        image: openebs/m-upgrade:ci
        volumeMounts:
        - name: config
          mountPath: /etc/config
          readOnly: true
        env:
        - name: OPENEBS_IO_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: OPENEBS_IO_POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
      volumes:
      - name: config
        configMap:
          name: pool-upgrade-config
      restartPolicy: Never
```

#### This is updated runtask for remote exec
```yaml
apiVersion: openebs.io/v1alpha1
kind: RunTask
metadata:
  name: check-quorum
  namespace: default
spec:
  meta: |
    id: quormCheck
    apiVersion: v1
    kind: Pod
    action: exec
    runNamespace: openebs
    objectName: pool-pod-ksk2c
  task: |-
    command:
    - zfs
    - get
    - quorum
    - -H
    - cstor-dnd3k
    - -o
    - value
    container: cstor-pool-mgmt
    stdin: false
    stdout: true
    stderr: true
  post: |-
    {{- jsonpath .JsonResult "{.stdout}" | trim | saveAs "quorumCheck.stdout" .TaskResult | noop -}}
```
Signed-off-by: Shovan Maity <shovan.maity@mayadata.io>

### CODE REVIEW

File: cmd/upgrade/app/v1alpha1/executor.go [hunk 1]
```diff
@@ -71,12 +67,20 @@ func ExecutorBuilderForConfig(cfg *apis.UpgradeConfig) *ExecutorBuilder {
...
 		executorBuilder.Errors = append(executorBuilder.Errors,
-			errors.Errorf("failed to instantiate executor builder: ENV {%s} not present", envSelfUID))
+			errors.Wrap(err, "failed to instantiate executor builder"))
 		return executorBuilder
+	}
 
+	owners := p.ObjectMeta.OwnerReferences
+	if len(owners) == 0 {
+		executorBuilder.Errors = append(executorBuilder.Errors,
+			errors.New("failed to instantiate executor builder:unable to find owner"))
```
Comment (AmitKumarDas) [inline cmd/upgrade/app/v1alpha1/executor.go]: can give a space after `:`
```go
errors.New("failed to instantiate executor builder: pod does not have a owner")
```

File: cmd/upgrade/app/v1alpha1/executor.go [hunk 2]
```diff
@@ -35,9 +34,6 @@ const (
 	// envSelfNamespace represent namespace of job in which upgrade process is running.
 	// This is required to build owner reference of upgrade result cr.
 	envSelfNamespace = "OPENEBS_IO_SELF_NAMESPACE"
```
Comment (AmitKumarDas) [inline cmd/upgrade/app/v1alpha1/executor.go]: lets rename SELF to POD in env variable name as well as code variable name

File: cmd/upgrade/app/v1alpha1/executor.go [hunk 3]
```diff
@@ -101,9 +105,8 @@ func ExecutorBuilderForConfig(cfg *apis.UpgradeConfig) *ExecutorBuilder {
 	for _, resource := range cfg.Resources {
 		resource := resource // pin it
 		upgradeResult, err := NewUpgradeResultGetOrCreateBuilder().
-			WithSelfName(selfName).
 			WithSelfNamespace(selfNamespace).
-			WithSelfUID(selfUID).
+			WithSelfOwner(&owners[0]).
```
Comment (AmitKumarDas) [inline cmd/upgrade/app/v1alpha1/executor.go]: Check if this can be `WithOwner`

File: cmd/upgrade/app/v1alpha1/result.go [hunk 4]
```diff
@@ -49,9 +46,8 @@ type UpgradeResult struct {
 // UpgradeResultGetOrCreateBuilder helps to get or create UpgradeResult instance
 type UpgradeResultGetOrCreateBuilder struct {
 	*errors.ErrorList
-	SelfName        string
 	SelfNamespace   string
-	SelfUID         types.UID
+	SelfOwner       *metav1.OwnerReference
```
Comment (AmitKumarDas) [inline cmd/upgrade/app/v1alpha1/result.go]: Can this be `Owner` ?
Can we write comments for each field?

File: cmd/upgrade/app/v1alpha1/executor.go (hunk 3 above)
Comment (imazik) [reply cmd/upgrade/app/v1alpha1/executor.go]: done

File: cmd/upgrade/app/v1alpha1/result.go (hunk 4 above)
Comment (imazik) [reply cmd/upgrade/app/v1alpha1/result.go]: done

File: cmd/upgrade/app/v1alpha1/executor.go (hunk 1 above)
Comment (imazik) [reply cmd/upgrade/app/v1alpha1/executor.go]: done

File: cmd/upgrade/app/v1alpha1/executor.go (hunk 2 above)
Comment (imazik) [reply cmd/upgrade/app/v1alpha1/executor.go]: done

File: cmd/upgrade/app/v1alpha1/executor.go [hunk 5]
```diff
@@ -19,25 +19,21 @@ package v1alpha1
...
-	// envSelfName represent name of job in which upgrade process is running.
-	// This is required to build owner reference of upgrade result cr.
-	envSelfName = "OPENEBS_IO_SELF_NAME"
-	// envSelfNamespace represent namespace of job in which upgrade process is running.
-	// This is required to build owner reference of upgrade result cr.
-	envSelfNamespace = "OPENEBS_IO_SELF_NAMESPACE"
-	// envSelfUID represent UID of job in which upgrade process is running.
-	// This is required to build owner reference of upgrade result cr.
-	envSelfUID = "OPENEBS_IO_SELF_UID"
+	// envSelfName represent name of pod in which upgrade process is running.
```
Comment (AmitKumarDas) [inline cmd/upgrade/app/v1alpha1/executor.go]: comments will need a change

File: cmd/upgrade/app/v1alpha1/executor.go [hunk 6]
```diff
@@ -19,25 +19,21 @@ package v1alpha1
...
+	// This is required to get owner reference for upgrade result cr.
+	envPodName = "OPENEBS_IO_POD_NAME"
+	// envSelfNamespace represent namespace of pod in which upgrade process is running.
```
Comment (AmitKumarDas) [inline cmd/upgrade/app/v1alpha1/executor.go]: comments will need a change

**Pull request merged**